include/itkAttenuationImageFilter.h

```diff
@@ -0,0 +1,204 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkAttenuationImageFilter_h
+#define itkAttenuationImageFilter_h
+
+#include "itkImage.h"
+#include "itkImageToImageFilter.h"
+#include "itkImageRegionSplitterDirection.h"
+#include "itkMacro.h"
+
+#include "itkNumericTraits.h"
+
+namespace itk
+{
+/** \class AttenuationImageFilter
+ * \brief Computes the estimated attentuation in dB/(MHz*cm)
+ *
+ * Image sink receives an image and outputs estimated attenuation.
+ *
+ * \sa VariableLengthVector
+ *
+ * \ingroup Operators
+ * \ingroup ITKImageStatistics
+ * \ingroup Ultrasound
+ */
+template <typename TInputImage,
+          typename TOutputImage = itk::Image<double, TInputImage::ImageDimension>,
```

Comment:
Shouldn't this be `float` by default? I am afraid that `double` would only waste memory and make memory operations slower. This data is expected to be quite noisy, and therefore we don't need high precision.

Reply:
> This data is expected to be quite noisy

This is the distinction I wasn't sure about, thanks. Will change to `float`.